genesyscloud/data_source_genesyscloud_routing_utilization_label.go

```diff
@@ -0,0 +1,53 @@
+package genesyscloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v119/platformclientv2"
+)
+
+func dataSourceRoutingUtilizationLabel() *schema.Resource {
+	return &schema.Resource{
+		Description: "Data source for Genesys Cloud Routing Utilization Labels. Select a label by name.",
+		ReadContext: ReadWithPooledClient(dataSourceRoutingUtilizationLabelRead),
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Label name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func dataSourceRoutingUtilizationLabelRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	sdkConfig := m.(*ProviderMeta).ClientConfig
+	routingAPI := platformclientv2.NewRoutingApiWithConfig(sdkConfig)
+
+	name := d.Get("name").(string)
+
+	return WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
+		labels, _, getErr := routingAPI.GetRoutingUtilizationLabels(1, 1, "", name)
+		if getErr != nil {
+			return retry.NonRetryableError(fmt.Errorf("Error requesting label %s: %s", name, getErr))
+		}
+
+		if labels.Entities == nil || len(*labels.Entities) == 0 {
+			return retry.RetryableError(fmt.Errorf("No labels found with name %s", name))
+		}
+
+		if len(*labels.Entities) == 1 {
+			label := (*labels.Entities)[0]
+			d.SetId(*label.Id)
+			return nil
+		}
+
+		return retry.NonRetryableError(fmt.Errorf("%d labels found with name %s. The data source might be misconfigured", len(*labels.Entities), name))
+	})
```

Comment:
If the api is given "abc", it will return exact match "abc" or nothing.

I added this because the API does support partial matching, using asterisks. So if given "*abc*" it would return "abcd" + whatever else.

My reasoning was if the person writing the data_source wrote "*abc*", then they'd be matching on unexpected things.

Could just throw an error if the input has * instead?

I'll make that change, let me know what you think.

Reply:
Ah okay, I'm used to endpoints implementing partial match no the name. Sounds good.